src/Data/TypeRepMap/Internal.hs

```diff
@@ -217,26 +217,46 @@ delete = fromTriples . deleteByFst (typeFp @a) . toTriples
 {-# INLINE delete #-}
 
 {- |
-Update a value at a specific key with the result of the provided function. When
-the key is not a member of the map, the original map is returned.
+Update a value at a specific key with the result of the provided function. 
+When the key is not a member of the map, the original map is returned.
 
 >>> trmap = fromList @(TypeRepMap Identity) [WrapTypeable $ Identity "a"]
 >>> lookup @String $ adjust (fmap (++ "ww")) trmap
 Just (Identity "aww")
 -}
 adjust :: forall a f . Typeable a => (f a -> f a) -> TypeRepMap f -> TypeRepMap f
-adjust fun tr = case cachedBinarySearch (typeFp @a) (fingerprintAs tr) (fingerprintBs tr) of
-    Nothing -> tr
-    Just i  -> tr {trAnys = changeAnyArr i (trAnys tr)}
+adjust fun tr = alter (fmap fun) tr
```

Comment:
Nice! This implementation became much shorter :slightly_smiling_face: Can also eta-reduce it a bit
```suggestion
adjust fun = alter (fmap fun)
```